spring-cloud-task-jdbchdfs-common/src/main/java/org/springframework/cloud/task/jdbchdfs/common/HdfsTextItemWriter.java

```diff
@@ -0,0 +1,232 @@
+import org.springframework.data.hadoop.store.strategy.naming.FileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.naming.RollingFileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.naming.StaticFileNamingStrategy;
+import org.springframework.data.hadoop.store.strategy.rollover.RolloverStrategy;
+import org.springframework.data.hadoop.store.strategy.rollover.SizeRolloverStrategy;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Writes items as a byte array to a destination as specified by the fsURI.
+ *
+ * @author Glenn Renfro
+ */
+public class HdfsTextItemWriter<T> extends AbstractItemStreamItemWriter<T> implements InitializingBean {
+
+	private static final String DEFAULT_LINE_SEPARATOR = System.getProperty("line.separator");
+
+	public static final String DEFAULT_FILENAME = "data";
+
+	public static final String DEFAULT_BASE_PATH = "/data/";
+
+	public static final String DEFAULT_FILE_EXTENSION = "log";
+
+	public static long DEFAULT_ROLLOVER_THRESHOLD_IN_BYTES = 10 * 1024 * 1024; // 10MB
+
+	private volatile String charset = "UTF-8";
+
+	protected final Logger logger = LoggerFactory.getLogger(getClass());
+
+	private String fileName = DEFAULT_FILENAME;
+
+	private String basePath = DEFAULT_BASE_PATH;
+
+	private String fileExtension = DEFAULT_FILE_EXTENSION;
+
+	private long rolloverThresholdInBytes = DEFAULT_ROLLOVER_THRESHOLD_IN_BYTES;
+
+	private String fsUri;
+
+	private volatile Configuration configuration;
+
+	private volatile DataStoreWriter<byte[]> storeWriter;
+
+	private LineAggregator<T> lineAggregator;
+
+	private String lineSeparator = DEFAULT_LINE_SEPARATOR;
+
+	public void afterPropertiesSet() throws Exception {
+		Assert.notNull(lineAggregator, "A LineAggregator must be provided.");
+		Configuration configurationToUse = null;
+		if (StringUtils.hasText(fsUri)) {
+			configurationToUse = new Configuration(configuration);
+			configurationToUse.set(CommonConfigurationKeysPublic.FS_DEFAULT_NAME_KEY, fsUri);
+		}
+		else {
+			configurationToUse = configuration;
+		}
+
+		List<FileNamingStrategy> strategies = new ArrayList<>();
+		strategies.add(new StaticFileNamingStrategy(fileName));
+		strategies.add(new RollingFileNamingStrategy());
+		strategies.add(new StaticFileNamingStrategy(fileExtension, "."));
+		ChainedFileNamingStrategy fileNamingStrategy = new ChainedFileNamingStrategy();
+		fileNamingStrategy.setStrategies(strategies);
+		RolloverStrategy rolloverStrategy = new SizeRolloverStrategy(rolloverThresholdInBytes);
+		Path baseDirPath = new Path(basePath);
+		OutputStreamWriter writer = new OutputStreamWriter(configurationToUse, baseDirPath, null);
+		writer.setInWritingSuffix(".tmp");
+		writer.setFileNamingStrategy(fileNamingStrategy);
+		writer.setRolloverStrategy(rolloverStrategy);
+		storeWriter = writer;
+		if (storeWriter instanceof InitializingBean) {
```

Comment:
Is this considered a good practice in Spring or should we expect that `storeWriter` is a bean and this is handled by the container?